executor/runtime/docker/docker.go

```diff
@@ -2524,6 +2528,25 @@ func (r *DockerRuntime) waitForTiniConnection(ctx context.Context, l *net.UnixLi
 	}
 }
 
+func (r *DockerRuntime) waitForTiniConnections(ctx context.Context, listeners map[string]*net.UnixListener) (map[string]*net.UnixConn, error) {
+	if listeners == nil {
```

Comment:
I'm not sure why this check makes any difference. Without it, this function would just return an empty map, which is equivalent to a nil map as far as reads are concerned.

Reply:
I agree